### PR TITLE
CUDA threads/block via config_str, git downloads

### PR DESCRIPTION
**Pull request by martinschwinzerl**
## TL;DR: 
Adds support for explicitly stating the default grid dimensions for cuda kernels during the creation of a track-job via a TOML encoded config_str. Provides a new method to sync local copies of external libraries with upstream git repositories. Fixes a bug introduced in #108 concerning disabling nodes for OpenCL. Passes all tests on my machine and should not produce any new conflicts

## Detailed Changes
- CudaTrackJob now allows to set a default value for the threads/block value via the `config_str` parameter during track_job creation. Cf. minimal python example:
```
default_threads_per_block = 256
default_track_threads_per_block = 512

config_str = f"""
cuda.threads_per_block = {default_threads_per_block}\r\n
cuda.track_threads_per_block = {default_track_threads_per_block}\r\n
"""    
track_job = pyst.CudaTrackJob(lattice, particles, 0, config_str=config_str)
```
This can be used to investigate the effects of the current fixed-size grid on the run-time performance of CUDA tracking.

- Introduce support for parsing [TOML](https://github.com/toml-lang/toml) config files and strings. To this end, the [toml11](https://github.com/ToruNiina/toml11) library is embedded and used. The `config_str` variable for the CudaTrackJob is the first use-case for this method, since most objects in SixTrackLib take a config_str during creation, further uses of this feature are anticipated

- There are problems with the download of external projects like the OpenCL headers in newer versions of CMake with the "works also on CMake < 3.11" approach that is currently employed. To not mandate very bleeding-edge versions of CMake that supports the new `FetchContent` approach, a function is provided in `cmake/SetupGit.cmake` to clone or optionally sync a locally cloned repository with upstream. The function `Git_sync_with_repo` tries to only fetch the project if the ref's differ and be configured to also track a branch different than master. 

- Fixes a bug in the `SIXTRACKLIB_DEVICES` parsing of devices if there are more than one devices available that match a certain token in the string

## Problems
On CMake >= 3.15, a problem occurs with the linking of the host interface to cuda Kernels. Cf  #109 for further details. If possible, please try to build with CMake < 3.15 until this is resolved.

### CODE REVIEW

**Comment by martinschwinzerl** [comment]
Updated/merged with PR #123 . Even though this PR is older, my suggestion would be to first merge #123 and then look into merging this. The toml11 integration and config_str handling included here would be required for further development of the run-time compilation configuration

Passes all tests on my workstation computer

**Comment by martinschwinzerl** [comment]
Re-merged with #132, seems to pass all tests on my machine

**Comment by martinschwinzerl** [comment]
Merged into Pr #132 -> close
